docs.feldera.com/docs/pipelines/latency.md

```diff
@@ -0,0 +1,211 @@
+
+**Reported as**: Histogram over the last 600 seconds or 10,000 batches (whichever is smaller).
+
+Note that the granularity of a batch is determined by the input connector.
+
+**Example:**
+
+```
+input_connector_completion_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="0"} 0
+<skipped>
+input_connector_completion_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="0.199999"} 13
+input_connector_completion_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="0.299999"} 40
+input_connector_completion_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="0.399999"} 54
+<skipped>
+input_connector_completion_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="inf"} 70
+input_connector_completion_latency_seconds_sum{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0"} 30.134436
+input_connector_completion_latency_seconds_count{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0"} 70
+```
+
+### Processing latency (`input_connector_processing_latency_seconds`)
+
+**Scope**: Per input connector
+
+**Definition**: Latency from ingestion of a batch until the SQL engine has computed all updates (before they are delivered to sinks).
+
+**Reported as**: Histogram over the last 600 seconds or 10,000 batches (whichever is smaller).
+
+**Example:**
+
+```
+input_connector_processing_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="0"} 0
+<skipped>
+input_connector_processing_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="0.199999"} 42
+input_connector_processing_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="0.299999"} 109
+input_connector_processing_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="0.399999"} 120
+<skipped>
+input_connector_processing_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="inf"} 177
+input_connector_processing_latency_seconds_sum{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0"} 138.936283
+input_connector_processing_latency_seconds_count{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0"} 177
+```
+
+### Step latency (`dbsp_step_latency_seconds`)
+
+**Scope**: Pipeline-wide
+
+**Definition**: Time the SQL engine takes to complete a single processing step.
```

Comment:
complete measured from ...
the processing step will process all the data in a batch for each input